CMSgov/design-system
# [WNMGDS-1641] Implement styled custom controls for DayPicker

## Summary

https://jira.cms.gov/browse/WNMGDS-1641

### Changes

- Added `CustomDayPicker` and `CustomDayPickerCaption`, which implement the custom controls
- Styled the new controls
- Styled the component's error state
- Made some changes to `useFormLabel` so that error-related info would be automatically passed down to the `TextInput`
- Created a Sass variable for the error border width, which we actually need to talk about because we recently filed a related bug
- Start exporting the Calendar icon, and include them in the example story
- Update visual regression tests

## How to test

Check out the stories and try all hover, focus, and active states, and interact with the new controls.

https://design-system-demo.s3.amazonaws.com/pwolfert/day-picker-custom-controls/core/storybook/index.html?path=/story/components-singleinputdatefield--with-picker
